app/src/main/java/org/mozilla/focus/widget/InlineAutocompleteEditText.java

```diff
@@ -646,19 +646,11 @@ private void setIsKeyFromRemoteAppOrSoftKeyboardClear(final boolean isKeyFromRem
     public boolean dispatchKeyEvent(KeyEvent event) {
         // This prevents the selector from entering the text area
         final int keyCode = event.getKeyCode();
-        final boolean shouldDispatch = keyCode != KeyEvent.KEYCODE_DPAD_DOWN
-                || keyCode != KeyEvent.KEYCODE_DPAD_DOWN_LEFT
-                || keyCode != KeyEvent.KEYCODE_DPAD_DOWN_RIGHT
-                || keyCode != KeyEvent.KEYCODE_DPAD_LEFT
-                || keyCode != KeyEvent.KEYCODE_DPAD_RIGHT
-                || keyCode != KeyEvent.KEYCODE_DPAD_UP
-                || keyCode != KeyEvent.KEYCODE_DPAD_UP_LEFT
-                || keyCode != KeyEvent.KEYCODE_DPAD_UP_RIGHT;
-
-        if (shouldDispatch) {
-            return super.dispatchKeyEvent(event);
-        }
+        final boolean shouldDispatch = !(keyCode == KeyEvent.KEYCODE_DPAD_DOWN
+                || keyCode == KeyEvent.KEYCODE_DPAD_LEFT
```

Comment:
Are we not worried about these other down-right, up-left keys anymore?

Reply:
@liuche I tried every way possible and noticed the combos never got hit. It always sent two events. Maybe there is a remote that has diagonal buttons? I removed it to simplify the code a bit

Reply:
The Android code is pretty generic and assumes you can have any device (e.g. a video game controller). I think ppl could pair bluetooth controllers to their devices and this combination could get hit – I think I'd prefer to leave it in. If we want to simplify the code, we could stick this in a helper function or use a range (since the DPAD_* are almost certainly ints that are all next to one another).